### PR TITLE
Makes assistants reasonably runnable

### PR DESCRIPTION
**Pull request by daveFNbuck**
This contains commits that allow the assistant to be run on the command-line, stay alive while waiting for new jobs to be scheduled, and have the scheduler properly recognize that the assistant is running jobs rather than marking them as failed. This provides basic functionality to run the assistant, but it doesn't tend to last long for me before it gets a job type it doesn't know about.

### CODE REVIEW

**Comment by Tarrasch** [comment]
Cool stuff! Excited to hear reviews from the community. I commited in your individual commits. :)

**Comment by erikbern** [comment]
This looks great!

Note that `interface.load_task` can load modules dynamically. If we keep track of the module of all tasks, and return it to the worker/assistant, we should be able to load modules dynamically and execute anything that's available to the assistant

**Comment by daveFNbuck** [comment]
It would be great if we added the module tracking, as my assistant currently fails fairly quickly because it doesn't load everything thoroughly enough. I'm working on making that report a failure (which is still necessary) but that would be a better fix for my actual problem. I'll dig into that when I'm done with the failures.

File: luigi/scheduler.py [hunk 1]
```diff
@@ -224,6 +224,9 @@ def prune(self, config):
         if self.last_active + config.worker_disconnect_delay < time.time():
             return True
 
+    def assistant(self):
```
Comment (berkerpeksag) [inline luigi/scheduler.py]: This could be a property. `worker.assistant` instead of `worker.assistant()` looks better to me :)

**Comment by erikbern** [comment]
LGTM!

File: luigi/scheduler.py (hunk 1 above)
Comment (Tarrasch) [inline luigi/scheduler.py]: I still haven't learned about the best practive for when to use properties in python or not. Hmm ...

Comment (daveFNbuck) [inline luigi/scheduler.py]: It seems reasonable here, as that gives us the option to make assistant a member variable in the future.

**Pull request merged**